shell_integration/windows/OCUtil/RemotePathChecker.cpp

```diff
@@ -172,10 +176,13 @@ RemotePathChecker::~RemotePathChecker()
     CloseHandle(_newQueries);
 }
 
-vector<wstring> RemotePathChecker::WatchedDirectories()
+void RemotePathChecker::UpdateWatchedDirectories(vector<wstring>& localWatchedDirectories, unsigned& localVersion) const
 {
+    if (localVersion == _watchedDirectoriesVersion)
```

Comment:
Without std::atomic, this is a undefined behavior according to the C++ standard.